RepositoryCompiler/CodeModel/CodeParsers/CSharp/CaDETMemberMetricCalculator.cs

```diff
@@ -121,5 +141,296 @@ private int CountOccurrences(string text, string pattern)
+                            n.IsKind(SyntaxKind.SubtractExpression) ||
+                            n.IsKind(SyntaxKind.MultiplyExpression) ||
+                            n.IsKind(SyntaxKind.DivideExpression) ||
+                            n.IsKind(SyntaxKind.ModuloExpression) ||
+                            n.IsKind(SyntaxKind.LeftShiftExpression) ||
+                            n.IsKind(SyntaxKind.RightShiftExpression));
+        }
+
+        private int CountUnaryExpressions(MemberDeclarationSyntax method)
+        {
+            int count = method.DescendantNodes().OfType<PrefixUnaryExpressionSyntax>()
+                .Count(n => n.IsKind(SyntaxKind.PreIncrementExpression) ||
+                            n.IsKind(SyntaxKind.PreDecrementExpression) ||
+                            n.IsKind(SyntaxKind.UnaryPlusExpression) ||
+                            n.IsKind(SyntaxKind.UnaryMinusExpression));
+            count += method.DescendantNodes().OfType<PostfixUnaryExpressionSyntax>()
+                .Count(n => n.IsKind(SyntaxKind.PostIncrementExpression) ||
+                            n.IsKind(SyntaxKind.PostDecrementExpression));
+            return count;
+        }
+
+        private int CountNumberOfParenthesizedExpressions(MemberDeclarationSyntax method)
+        {
+            return method.DescendantNodes().OfType<ExpressionSyntax>()
+                .Count(n => n.IsKind(SyntaxKind.ParenthesizedExpression));
+        }
+
+        private int CountNumberOfLambdaExpressions(MemberDeclarationSyntax method)
+        {
+            return method.DescendantNodes().OfType<LambdaExpressionSyntax>().Count();
+        }
+
+        private int CountMaxNestedBlocks(MemberDeclarationSyntax method)
+        {
+            var blocks = method.DescendantNodes().OfType<BlockSyntax>().ToList();
+
+            List<int> blocksAncestors = new List<int>();
+            foreach (var block in blocks)
+            {
+                blocksAncestors.Add(block.Ancestors().Count(a => a.IsKind(SyntaxKind.Block)));
+            }
+
+            if (!blocksAncestors.Any()) return 0;
+            return blocksAncestors.Max();
+        }
+
+        private int CountNumberOfUniqueWords(MemberDeclarationSyntax method)
+        {
+            if (!method.Kind().Equals(SyntaxKind.MethodDeclaration)) return 0;
+
+            BaseMethodDeclarationSyntax baseMethod = (BaseMethodDeclarationSyntax)method;
+            string methodBody = RemoveCommentsFromCode(baseMethod.Body);
+            methodBody = RemoveSymbols(methodBody);
+            List<string> words = GetWords(methodBody);
+            words = BreakWords(words);
+            words = FilterWords(words);
+            return words.Distinct().Count();
+        }
+
+        private List<string> GetWords(string methodBody)
+        {
+            List<string> words = new List<string>();
+            
+            int index = 0;
+            while (index < methodBody.Length && char.IsWhiteSpace(methodBody[index]))
```

Comment:
Can we not replace this code by calling split against " " (space), "\n" and "\r" and trimming each element?